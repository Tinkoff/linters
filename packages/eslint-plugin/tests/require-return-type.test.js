const { CLIEngine } = require('eslint');
const path = require('path');

const cli = new CLIEngine({
  color: false,
  envs: ['browser', 'node'],
  useEslintrc: false,
  configFile: require.resolve(
    '@tinkoff/linters/eslint/common/typescript/enviroment.js'
  ),
  parserOptions: {
    createDefaultProgram: true,
  },
  rules: {
    '@tinkoff/require-return-type': 'error',
  },
  ignore: false,
});

const codeframe = cli.getFormatter('codeframe');

test('require-return-type snapshot', () => {
  const report = cli.executeOnFiles([
    path.join(__dirname, './require-return-type.fixture.ts'),
  ]);

  expect(codeframe(report.results)).toMatchSnapshot();
});
