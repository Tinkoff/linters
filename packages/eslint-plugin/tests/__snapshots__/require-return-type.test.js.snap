// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-return-type snapshot 1`] = `""`;
